File: leetcode/leetcode_104_递归.py
# python2 递归解题思路 -- 深度优先搜索
class treeNode(object):
        def __init__(self, x):
                self.val = x;
                self.left = None
                self.right = None


class Solution:
        def maxDepth(self, root):
                if root is None:
                        return 0
                else:
                        left_hight = self.maxDepth(root.left)
                        right_hight = self.maxDepth(root.right)
                        return max(left_hight, right_hight) + 1

File: leetcode/leetcode_104_递归.cpp
class Solution {
public:
    int maxDepth(TreeNode* root) {
        if(root == nullptr)
            return 0;
        TreeNode* ptr = root;
        TreeNode* left_ptr = ptr->left;
        TreeNode* right_ptr = ptr->right;
        int left_depth = maxDepth(left_ptr);
        int right_depth = maxDepth(right_ptr);
        if (left_depth > right_depth)
            return left_depth + 1;
        else
            return right_depth + 1;
    }
};
